lua/nvim-treesitter/configs.lua

```diff
@@ -218,28 +220,28 @@ local config = {
       enable = false,
       disable = {},
       keymaps = {
+        init_selection="gnn",
         node_incremental="grn",
-        scope_incremental="grc"
+        scope_incremental="grc",
+        node_decremental="grm"
       },
-      is_supported = function() return true end
+      is_supported = function(ft)
+        return queries.get_query(ft, 'locals')
+      end
     },
     node_movement = {
       enable = false,
       disable = {},
-      is_supported = function() return true end,
+      is_supported = function(ft)
+        return queries.get_query(ft, 'locals')
+      end,
       keymaps = {
-        move_up = "<a-k>",
-        move_down = "<a-j>",
-        move_left = "<a-h>",
-        move_right = "<a-l>",
+        parent_scope = "tk",
+        child_scope = "tj",
+        next_scope = "tl",
+        previous_scope = "th",
       },
```

Comment:
Well I tried to use the node movement, and it really made no sense to me, i am used to move around with the regular vim motions which is still the best way to move in vim imho. 

But in most languages (aka lua, c, rust, c++, java,(*script)...) you are moving a lot between scopes. I'm not a python developer, but it seems to me that `if` and `for` still kind of represent scope, delimited by tabulations though.

Maybe node movement would be better for something like switching nodes as you showed earlier. That's why i left those functions around.

The issue with the nodes representing multiple nodes at the same cursor position is actually pretty disturbing, that's why i decided to just use the smallest named node under the cursor, so we have more control over what we want to do. Because you can expand from there to a bigger scope.

But i'd have to say, even with the scope movement, i find it a little bit hard to understand how to move as quickly as i'm used to.

Reply:
Ok. I get that. Then I move that stuff simply to my personal configuration.

I made Python locals.scm where I cheated also a little bit and made `for`/`while` scopes. Tt's possible but then you have to be careful whether people define scope by something they would like to move around or by the actual scoping rules variable.

Reply:
Maybe it's then better to define movements for specific syntax nodes then? Like `next_function`/`previous_function`... vim has `]]` but that's only working on text not the real syntax.

Reply:
i actually thought about this. Scope movement is quite disturbing, you never really know where you are going.

I think that instead of providing the bindings ourselves, we should let the users decide and provide utility functions so they can basically do what they want with it. 
I already started stabilizing the api with this PR, i will work on providing this api as simple as possible in my next PR.

Reply:
Pretty low level concern : the `t...` mappings are already used by default neovim. We may want to keep the previous ones.